include/tarxx.h

```diff
@@ -292,18 +361,93 @@ namespace tarxx {
             return str;
         }
 
+        void init_lz4()
+        {
+            if (compression_ != compression_mode::lz4) {
+                return;
+            }
+
+            lz4_ctx_ = std::make_unique<lz4_ctx>();
+            const auto outbuf_size = lz4_check_error(LZ4F_compressBound, 16 * 1024, &lz4_prefs_);
+
+            lz4_out_buf_.reserve(outbuf_size);
+            const auto headerSize = lz4_check_error(LZ4F_compressBegin, lz4_ctx_->get(), lz4_out_buf_.data(), lz4_out_buf_.capacity(), &lz4_prefs_);
+            lz4_out_buf_pos_ += headerSize;
+            write_lz4_data();
+        }
+
+        void write_lz4_data()
+        {
+            unsigned long long offset = 0;
+            switch (mode_) {
+                // guarding that input is not stream is done in add_file_streaming
+                // using a file as input is fine, as we have all data necessary
+                // for compression w/o seeking
+                case output_mode::stream_output:
+                    while (lz4_out_buf_pos_ > 0) {
+                        block_t block {};
+                        const auto copy_size = std::min(lz4_out_buf_pos_, block.size());
+                        std::copy_n(lz4_out_buf_.begin() + offset, copy_size, block.data());
+                        callback_(block, copy_size);
+                        lz4_out_buf_pos_ -= copy_size;
+                        offset += copy_size;
+                    }
+
+                    break;
+                case output_mode::file_output:
+                    file_.write(lz4_out_buf_.data(), lz4_out_buf_pos_);
+                    lz4_out_buf_pos_ = 0;
+                    break;
+            }
+        }
```

Comment:
please only compile in if lz4 support is activated (see comments in `CMakeLists`)